ITEP-68044 Live inference using camera

dwesolow opened the pull request:
## 📝 Description

The goal of this PR is to make the inference using camera more UX friendly, i.e. reduce the steps to achieve the end goal.
We decided to separate inference into two modes: inference on file and inference using screenshots from the camera.

In the next PR I'll remove outdated code regarding camera inference from `quick inference` and `isDisabled` from `QuickInferenceProvider`.

JIRA: ITEP-68044

https://github.com/user-attachments/assets/84385977-0076-4a72-af91-2e255b30a9eb

<!--  
If the PR addresses a specific GitHub issue, include one of the following lines to enable auto-closing:
Fixes #<issue_number>
Closes #<issue_number>

If referencing an internal ticket (e.g. JIRA), include the ticket number instead:
JIRA: <project-key>-<ticket-number>

If there’s no related issue or ticket, you can skip this section.
-->

## ✨ Type of Change

Select the type of change your PR introduces:

- [X] 🔨 **Refactor** – Non-breaking change which refactors the code base
- [X] 🧪 **Tests**

## 🧪 Testing Scenarios

Describe how the changes were tested and how reviewers can test them too:

- [X] ✅ Tested manually
- [ ] 🤖 Run automated end-to-end tests


## ✅ Checklist

Before submitting the PR, ensure the following:

- [X] 🔍 PR title is clear and descriptive
- [X] 📝 For internal contributors: If applicable, include the JIRA ticket number (e.g., ITEP-123456) in the PR **title**. Do **not** include full URLs
- [ ] 💬 I have commented my code, especially in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [X] ✅ I have added tests that prove my fix is effective or my feature works

jpggvilaca commented on web_ui/src/pages/project-details/components/project-tests/quick-inference/live-camera-inference/live-camera-inference.component.tsx:
```diff
@@ -0,0 +1,199 @@
+    const { image, onResetImage } = useQuickInference();
+    const isInferencedImageVisible = image !== undefined;
+
+    const handleTakeNext = () => {
```
np: we only use it once and without any args, can we not use `onResetImage` directly?

dwesolow replied:
Ah, right. Updated.

MarkRedeman commented on web_ui/packages/ui/src/toggle-buttons/toggle-buttons.component.tsx:
```diff
@@ -21,6 +21,7 @@ const ToggleButton = <T extends string>({
 }: ToggleButtonProps<T>) => {
     return (
         <Button
+            data-activated={selectedOption === option}
```
Alternatively we might also be able to use `aria-selected` or `aria-pressed`, or maybe use [ToggleButton](https://react-spectrum.adobe.com/react-spectrum/ToggleButton.html)?

camiloHimura commented on web_ui/src/pages/camera-page/camera-page.component.tsx:
```diff
@@ -35,8 +39,8 @@ export const CameraPage = (): JSX.Element => {
 
-    const isPermissionDenied = hasPermissionsDenied(userPermissions);
-    const isPermissionPending = userPermissions === UserCameraPermission.PENDING;
+    const permissionDenied = hasPermissionsDenied(userPermissions);
```
I prefer the original name, the new one doesn’t clearly indicate that it’s a boolean.

camiloHimura commented:
The 'square' style looks a bit off to me, since all buttons and switch components in the app are rounded, I think this should follow the same design for consistency.

![image](https://github.com/user-attachments/assets/0f0b3f09-2254-429e-80bb-0e211a88a421)

camiloHimura commented:
On large screens, the camera preview doesn't utilize all the available space, even with the sidebar collapsed. Also, the collapse button should have a darker color and a smaller icon (Check out the annotator and camera page sidebar collapsing for reference)

![Screenshot 2025-06-06 at 08 47 58](https://github.com/user-attachments/assets/b160cf00-d15a-4d97-b607-815edf738824)

dwesolow commented:
> The 'square' style looks a bit off to me, since all buttons and switch components in the app are rounded, I think this should follow the same design for consistency.
> 
> ![image](https://private-user-images.githubusercontent.com/6851052/452186130-0f0b3f09-2254-429e-80bb-0e211a88a421.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxOTQ2MjMsIm5iZiI6MTc0OTE5NDMyMywicGF0aCI6Ii82ODUxMDUyLzQ1MjE4NjEzMC0wZjBiM2YwOS0yMjU0LTQyOWUtODBiYi0wZTIxMWE4OGE0MjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMDcxODQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzRjZmRlODkxM2Y4NDQ2Njk4MzIwY2ExM2M0ZGU1Nzg2MTk3ZWM3YjJlNzJlYzk5ODkwYTM1YWZkZjg3YzUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.p4aHa9tzO3knCgWXD3VJFPWVJKrh0so2rwF9Gsunxjk)

To me it looks fine. I talked with Mariusz and he agreed with my proposal.
BTW this toggle buttons were not introduced in the current PR, we have it in the codebase. Rounded toggle buttons would look really weird.

dwesolow commented:
> Since the user can’t change the camera config and the only available actions are taking another shot or zooming in/out, I think the sidebar could be collapsed. Also, `Take Next Shot` would be more descriptive; `Take Next` feels too generic.
> 
> ![image](https://private-user-images.githubusercontent.com/6851052/452191941-6fa6ddfe-7beb-41e3-b102-84d44f8f0931.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxOTQ2MjMsIm5iZiI6MTc0OTE5NDMyMywicGF0aCI6Ii82ODUxMDUyLzQ1MjE5MTk0MS02ZmE2ZGRmZS03YmViLTQxZTMtYjEwMi04NGQ0NGY4ZjA5MzEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMDcxODQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGEwY2QwYjAyNjE3NjJjODhjNjNmYTg4NjU1OGIwMmQyYTg2MGExYWY3YTdkMmMyYmY3MzJiZGIwNTM1NjFkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.kcVYJtKnKH-mnQolQnD1OwiXp51q2amdgV0hBHkZ0N0)

Our goal was to allow user to collapse/show the sidebar and not introduce layout shift while switching between camera preview view and prediction view. If we collapsed the sidebar automatically then we would introduce unnecessary layout shifts. If user is not interested in the camera settings, they can collapse it.

camiloHimura commented:
> > Since the user can’t change the camera config and the only available actions are taking another shot or zooming in/out, I think the sidebar could be collapsed. Also, `Take Next Shot` would be more descriptive; `Take Next` feels too generic.
> > ![image](https://private-user-images.githubusercontent.com/6851052/452191941-6fa6ddfe-7beb-41e3-b102-84d44f8f0931.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxOTQ2MjMsIm5iZiI6MTc0OTE5NDMyMywicGF0aCI6Ii82ODUxMDUyLzQ1MjE5MTk0MS02ZmE2ZGRmZS03YmViLTQxZTMtYjEwMi04NGQ0NGY4ZjA5MzEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMDcxODQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGEwY2QwYjAyNjE3NjJjODhjNjNmYTg4NjU1OGIwMmQyYTg2MGExYWY3YTdkMmMyYmY3MzJiZGIwNTM1NjFkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.kcVYJtKnKH-mnQolQnD1OwiXp51q2amdgV0hBHkZ0N0)
> 
> Our goal was to allow user to collapse/show the sidebar and not introduce layout shift while switching between camera preview view and prediction view. If we collapsed the sidebar automatically then we would introduce unnecessary layout shifts. If user is not interested in the camera settings, they can collapse it.

I disagree, the it

camiloHimura commented:
> > On large screens, the camera preview doesn't utilize all the available space, even with the sidebar collapsed. Also, the collapse button should have a darker color and a smaller icon (Check out the annotator and camera page sidebar collapsing for reference)
> > ![Screenshot 2025-06-06 at 08 47 58](https://private-user-images.githubusercontent.com/6851052/452189260-b160cf00-d15a-4d97-b607-815edf738824.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxOTQ2MjMsIm5iZiI6MTc0OTE5NDMyMywicGF0aCI6Ii82ODUxMDUyLzQ1MjE4OTI2MC1iMTYwY2YwMC1kMTVhLTRkOTctYjYwNy04MTVlZGY3Mzg4MjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMDcxODQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDk3ODkzZjNiYzU3ZmRmMDI0YmJhY2M4MWRkYjI1ZWZiZDY4NTc5ZDkyODdmMmZiYTUyMjYxYTQzMTNmNjRhZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fGxdEJL-6xeCda-4LNdt7PCI5U9XgiDDawZPI07CvNE)
> 
> Why the icon should be smaller and darker? I don't set the width for the camera preview, user can change it in the camera settings I think.

Because that’s how the other sidebars render it, this one looks off in comparison.
![Screenshot 2025-06-06 at 10 57 13](https://github.com/user-attachments/assets/133ed863-1fc4-4bdf-b5ac-6b3deae670b0)
![Screenshot 2025-06-06 at 10 57 44](https://github.com/user-attachments/assets/aae94275-b2ff-4fce-83ee-712eebf565b0)

I wasn't referring to the camera feed configuration, but specifically how it’s rendered in the container; it should grow or shrink to fill all the available space.

camiloHimura commented:
> > The 'square' style looks a bit off to me, since all buttons and switch components in the app are rounded, I think this should follow the same design for consistency.
> > ![image](https://private-user-images.githubusercontent.com/6851052/452186130-0f0b3f09-2254-429e-80bb-0e211a88a421.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxOTQ2MjMsIm5iZiI6MTc0OTE5NDMyMywicGF0aCI6Ii82ODUxMDUyLzQ1MjE4NjEzMC0wZjBiM2YwOS0yMjU0LTQyOWUtODBiYi0wZTIxMWE4OGE0MjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMDcxODQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzRjZmRlODkxM2Y4NDQ2Njk4MzIwY2ExM2M0ZGU1Nzg2MTk3ZWM3YjJlNzJlYzk5ODkwYTM1YWZkZjg3YzUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.p4aHa9tzO3knCgWXD3VJFPWVJKrh0so2rwF9Gsunxjk)
> 
> To me it looks fine. I talked with Mariusz and he agreed with my proposal. BTW this toggle buttons were not introduced in the current PR, we have it in the codebase. Rounded toggle buttons would look really weird.

I don’t think it looks quite right, we don’t use that styling anywhere else in the app.
@jpggvilaca @MarkRedeman, what are your thoughts?

dwesolow commented:
> > > The 'square' style looks a bit off to me, since all buttons and switch components in the app are rounded, I think this should follow the same design for consistency.
> > > ![image](https://private-user-images.githubusercontent.com/6851052/452186130-0f0b3f09-2254-429e-80bb-0e211a88a421.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxOTQ2MjMsIm5iZiI6MTc0OTE5NDMyMywicGF0aCI6Ii82ODUxMDUyLzQ1MjE4NjEzMC0wZjBiM2YwOS0yMjU0LTQyOWUtODBiYi0wZTIxMWE4OGE0MjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMDcxODQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzRjZmRlODkxM2Y4NDQ2Njk4MzIwY2ExM2M0ZGU1Nzg2MTk3ZWM3YjJlNzJlYzk5ODkwYTM1YWZkZjg3YzUyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.p4aHa9tzO3knCgWXD3VJFPWVJKrh0so2rwF9Gsunxjk)
> > 
> > 
> > To me it looks fine. I talked with Mariusz and he agreed with my proposal. BTW this toggle buttons were not introduced in the current PR, we have it in the codebase. Rounded toggle buttons would look really weird.
> 
> I don’t think it looks quite right, we don’t use that styling anywhere else in the app. @jpggvilaca @MarkRedeman, what are your thoughts?

We use that in the new training dialog.

dwesolow commented:
> > > On large screens, the camera preview doesn't utilize all the available space, even with the sidebar collapsed. Also, the collapse button should have a darker color and a smaller icon (Check out the annotator and camera page sidebar collapsing for reference)
> > > ![Screenshot 2025-06-06 at 08 47 58](https://private-user-images.githubusercontent.com/6851052/452189260-b160cf00-d15a-4d97-b607-815edf738824.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxOTQ2MjMsIm5iZiI6MTc0OTE5NDMyMywicGF0aCI6Ii82ODUxMDUyLzQ1MjE4OTI2MC1iMTYwY2YwMC1kMTVhLTRkOTctYjYwNy04MTVlZGY3Mzg4MjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMDcxODQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDk3ODkzZjNiYzU3ZmRmMDI0YmJhY2M4MWRkYjI1ZWZiZDY4NTc5ZDkyODdmMmZiYTUyMjYxYTQzMTNmNjRhZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fGxdEJL-6xeCda-4LNdt7PCI5U9XgiDDawZPI07CvNE)
> > 
> > 
> > Why the icon should be smaller and darker? I don't set the width for the camera preview, user can change it in the camera settings I think.
> 
> Because that’s how the other sidebars render it, this one looks off in comparison. ![Screenshot 2025-06-06 at 10 57 13](https://private-user-images.githubusercontent.com/6851052/452232552-133ed863-1fc4-4bdf-b5ac-6b3deae670b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkyMDQ3NjEsIm5iZiI6MTc0OTIwNDQ2MSwicGF0aCI6Ii82ODUxMDUyLzQ1MjIzMjU1Mi0xMzNlZDg2My0xZmM0LTRiZGYtYjVhYy02YjNkZWFlNjcwYjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMTAwNzQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmEzMWI4OWFiZGM0ZDE5ZDhmYTQxYmJjODlmMzNlMmE3MWVhYmE3MWQ4NTMzYmRlZGI2OThmYTg4NjY2MmRmYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.zsfUnVXmRGfVtfajIFSNRVqKsYVuYW4aArHMCe1UVgA) ![Screenshot 2025-06-06 at 10 57 44](https://private-user-images.githubusercontent.com/6851052/452232617-aae94275-b2ff-4fce-83ee-712eebf565b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkyMDQ3NjEsIm5iZiI6MTc0OTIwNDQ2MSwicGF0aCI6Ii82ODUxMDUyLzQ1MjIzMjYxNy1hYWU5NDI3NS1iMmZmLTRmY2UtODNlZS03MTJlZWJmNTY1YjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA2MDZUMTAwNzQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDFlODU5NDRkODNmMmNhNWZiZjRiMGQ0Yjg3MDQ4OWYyODdjZmIyMmZkYmRjZjFmZDJhOWRjNGY5ZGQwZjk1NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.JSKpDG8N8q_uJg_skkocHrvBlbrwS0JXrnSoBJC8-GE)
> 
> I wasn't referring to the camera feed configuration, but specifically how it’s rendered in the container; it should grow or shrink to fill all the available space.

Updated:
- toggle buttons;
- toggle sidebar button to have the same styles like in camera upload;
- renamed `Take next` to `Take next shot`;
- camera preview to take all available space.

![image](https://github.com/user-attachments/assets/33d1b15d-01fe-48d8-87f1-d316205af5b7)
